### PR TITLE
Added documentation and Travis script for building on MacOS

### PR DESCRIPTION
**Pull request by ISNIT0**

### CODE REVIEW

**Comment by ISNIT0** [comment]
@mgautierfr Could you set up signing and distribution please?

File: travis/compile.sh [hunk 1]
```diff
@@ -25,3 +25,13 @@ fi
 meson . build ${MESON_OPTION}
 cd build
 ninja
+
+# Generate DMG
+if [[ "${TRAVIS_OS_NAME}" == "osx" ]]
+then
+  cd ..
+  mkdir zimwriterfs
+  cp -r build/src/* ./zimwriterfs
```
Comment (mgautierfr) [inline travis/compile.sh]: Why copy all the src content ?
We only need the `zimwriterfs` binary no ?
But, we are compiling dynamically on osx, so we probably need to package the library dependencies also.
I'm not sure how appdmg does it (if it does it)

File: appdmg.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
+{
+    "title": "ZimwriterFS Binary",
+    "icon": "openzim.png",
+    "window": {
+        "size": {
+            "width": 640,
+            "height": 344
+        }
+    },
```
Comment (mgautierfr) [inline appdmg.json]: zimwriterfs is a console application, what the `window` is for ?
(same question for the `x` and `y` above)

**Review by mgautierfr — CHANGES_REQUESTED** [review]
I'm a bit disturbed by this.
Historically, this is the travis in `kiwix-build` that publish things. The individual project's CIs are used to ensure the a change in this project is correct.
If we publish things in `zimwriterfs` CI, we will not be able to publish a new build version of `zimwriterfs` in we change `libzim` only. (And `zimwriterfs` CI has no access to ssh keys).

**Comment by mgautierfr** [comment]
Have you seen this http://mesonbuild.com/Creating-OSX-packages.html ?

**Comment by kelson42** [comment]
@ISNIT0 Would you be able to work with Matthieu to polish this PR and get it merged?

**Pull request closed without merging**